include/nuttx/fs/fs.h

```diff
@@ -206,16 +207,21 @@ struct file_operations
    * treated like unions.
    */
 
-  int     (*close)(FAR struct file *filep);
-  ssize_t (*read)(FAR struct file *filep, FAR char *buffer, size_t buflen);
-  ssize_t (*write)(FAR struct file *filep, FAR const char *buffer,
-                   size_t buflen);
-  off_t   (*seek)(FAR struct file *filep, off_t offset, int whence);
-  int     (*ioctl)(FAR struct file *filep, int cmd, unsigned long arg);
+  int      (*close)(FAR struct file *filep);
+  ssize_t  (*read)(FAR struct file *filep, FAR char *buffer, size_t buflen);
+  ssize_t  (*write)(FAR struct file *filep, FAR const char *buffer,
+                    size_t buflen);
+  off_t    (*seek)(FAR struct file *filep, off_t offset, int whence);
+  int      (*ioctl)(FAR struct file *filep, int cmd, unsigned long arg);
+  int      (*truncate)(FAR struct file *filep, off_t length);
+  FAR void *(*mmap)(FAR struct file *filep, off_t start, size_t length);
```

Comment:
how mmap return the different error number?

Reply:
To the user side, it is in errno. Of course in this interface we can also return differerent error numbers and pass the pointer in argument list. I was thinking that there is no use for other return values and this approach was simpler. Also user-side mmap only returns either address or MAP_FAILED.

Would you prefer and int type return and pointer in arguments list?

Reply:
> To the user side, it is in errno. Of course in this interface we can also return differerent error numbers and pass the pointer in argument list. I was thinking that there is no use for other return values and this approach was simpler. Also user-side mmap only returns either address or MAP_FAILED.
> 

but spec list the possible errno when when mmap return MAP_FAILD:
https://man7.org/linux/man-pages/man2/mmap.2.html

> Would you prefer an int type return and pointer in arguments list?

Yes, since only driver could know the real reason why mmap fail.

Reply:
Thats right! I will change it

Reply:
Done